crates/sui-graphql-rpc/src/types/dry_run_result.rs

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use super::base64::Base64;
+use super::move_type::MoveType;
+use super::transaction_block::TransactionBlock;
+use super::transaction_block_kind::programmable::TransactionArgument;
+use crate::error::Error;
+use async_graphql::*;
+use sui_json_rpc_types::{DevInspectResults, SuiExecutionResult};
+use sui_types::effects::TransactionEffects as NativeTransactionEffects;
+use sui_types::event::Event as NativeEvent;
+use sui_types::transaction::TransactionData as NativeTransactionData;
+
+#[derive(Clone, Debug, PartialEq, Eq, SimpleObject)]
+#[graphql(complex)]
+pub(crate) struct DryRunResult {
+    /// The error that occurred during dry run execution, if any.
+    pub error: Option<String>,
+    /// The intermediate results of the dry run execution.
+    pub results: Option<Vec<DryRunEffect>>,
+
+    #[graphql(skip)]
+    pub events: Vec<NativeEvent>,
+
+    #[graphql(skip)]
+    pub effects: NativeTransactionEffects,
+
+    #[graphql(skip)]
+    pub tx_data: NativeTransactionData,
+}
+
+#[ComplexObject]
+impl DryRunResult {
+    /// The transaction block representing the dry run execution.
+    pub async fn transaction(&self) -> Option<TransactionBlock> {
+        Some(TransactionBlock::DryRun {
+            tx_data: self.tx_data.clone(),
+            effects: self.effects.clone(),
+            events: self.events.clone(),
+        })
+    }
+}
+
+impl TryFrom<DevInspectResults> for DryRunResult {
+    type Error = crate::error::Error;
+    fn try_from(results: DevInspectResults) -> Result<Self, Self::Error> {
+        let execution_results = results
+            .results
+            .ok_or(Error::Internal(
+                "No execution results returned from dev inspect".to_string(),
+            ))?
+            .into_iter()
+            .map(DryRunEffect::try_from)
+            .collect::<Result<Vec<_>, Self::Error>>()?;
+        let events = results
+            .events
+            .data
+            .into_iter()
+            .map(|e| NativeEvent {
+                sender: e.sender,
+                package_id: e.package_id,
+                transaction_module: e.transaction_module,
+                type_: e.type_,
+                contents: e.bcs,
+            })
+            .collect();
+        let effects: NativeTransactionEffects =
+            bcs::from_bytes(&results.raw_effects).map_err(|e| {
+                Error::Internal(format!("Unable to deserialize transaction effects: {e}"))
+            })?;
+        let tx_data: NativeTransactionData = bcs::from_bytes(&results.raw_txn_data)
+            .map_err(|e| Error::Internal(format!("Unable to deserialize transaction data: {e}")))?;
+        Ok(Self {
+            error: results.error,
+            results: Some(execution_results),
+            events,
+            effects,
+            tx_data,
+        })
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, SimpleObject)]
+pub(crate) struct DryRunEffect {
```

Comment:
nit: Can you order the file to match the following pattern:

- Type definitions
- `#[Object]` and `#[ComplexObject]` impl blocks.
- Non-trait impl blocks
- trait impl blocks (I tend to put more domain-specific/special traits higher up).
- free functions

And in each section, start with the `pub` elements, then `pub(crate)`, then private?